### PR TITLE
BUG: implement .sort_index(...inplace=True) for #11402

### LINKED ISSUE #11402: Series.sort_index(inplace=True) always returns a new Series
Series.index_sort() _always_ returns a new Series, ignoring the `inplace` keyword argument. This is in contrast to DataFrame's sort_index() which works as expected.

```python
vals = range(10, 0, -1)
x = pd.Series(vals)

# Returns a new Series with the sorted index instead of
# changing x in place
y = x.sort_index(ascending=False, inplace=True)

# Not the same
(x != y).all()

# sort_index() works as expected for DataFrames
xx = pd.DataFrame({'col': x})
assert xx.sort_index(ascending=False, inplace=True) is None
```

show_versions() output:

INSTALLED VERSIONS
------------------
commit: None
python: 3.4.3.final.0
python-bits: 64
OS: Darwin
OS-release: 14.5.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8

pandas: 0.17.0
nose: 1.3.7
pip: 7.1.2
setuptools: 18.4
Cython: None
numpy: 1.10.1
scipy: 0.16.0
statsmodels: 0.6.1
IPython: 4.0.0
sphinx: None
patsy: 0.4.0
dateutil: 2.4.1
pytz: 2015.6
blosc: None
bottleneck: None
tables: 3.2.2
numexpr: 2.4.4
matplotlib: 1.4.3
openpyxl: 2.2.0-b1
xlrd: 0.9.3
xlwt: None
xlsxwriter: None
lxml: None
bs4: None
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.3
pymysql: 0.6.6.None
psycopg2: None

### PR DESCRIPTION
**Pull request by cfperez**
closes #11402

Adds inplace sort_index() for Series.

### CODE REVIEW

File: pandas/tests/test_series.py [hunk 1]
```diff
@@ -5328,6 +5328,26 @@ def test_sort_index(self):
         assert_series_equal(sorted_series,
                             self.ts.reindex(self.ts.index[::-1]))
 
+    def test_sort_index_inplace(self):
+        rindex = list(self.ts.index)
```
Comment (jreback) [inline pandas/tests/test_series.py]: add the issue number as a comment

File: pandas/tests/test_series.py [hunk 2]
```diff
@@ -5328,6 +5328,26 @@ def test_sort_index(self):
...
+        random.shuffle(rindex)
+
+        # descending
+        random_order = self.ts.reindex(rindex)
+        result = random_order.sort_index(ascending=False, inplace=True)
+        self.assertIs(result, None,
+                      msg='sort_index() inplace should return None')
+        assert_index_equal(random_order.index,
```
Comment (jreback) [inline pandas/tests/test_series.py]: compare a construsted series here (and not just the index)

**Comment by jreback** [comment]
pls also add a whatsnew note for 0.17.1 (put in API changes even though this is a bug-fix, as it technically is an api change)

File: pandas/tests/test_series.py (hunk 2 above)
Comment (cfperez) [inline pandas/tests/test_series.py]: Added index and series comparisons.

**Comment by cfperez** [comment]
Updated and running green.

**Comment by jreback** [comment]
merged via https://github.com/pydata/pandas/commit/1135ce34fc2fb995e239a7d36aee1b687ef69dd9

thanks!

**Pull request closed without merging**